src/hotspot/share/opto/superword.cpp

```diff
@@ -1630,36 +1621,88 @@ uint SuperWord::max_implemented_size(const Node_List* pack) {
   }
 }
 
-// Java API for Long.bitCount/numberOfLeadingZeros/numberOfTrailingZeros
-// returns int type, but Vector API for them returns long type. To unify
-// the implementation in backend, superword splits the vector implementation
-// for Java API into an execution node with long type plus another node
-// converting long to int.
-bool SuperWord::requires_long_to_int_conversion(int opc) {
-  switch(opc) {
-    case Op_PopCountL:
-    case Op_CountLeadingZerosL:
-    case Op_CountTrailingZerosL:
-      return true;
-    default:
-      return false;
+// If the j-th input for all nodes in the pack is the same unique input: return it, else nullptr.
+Node* PackSet::isa_unique_input_or_null(const Node_List* pack, int j) const {
+  Node* p0 = pack->at(0);
+  Node* unique = p0->in(j);
```

Comment:
Could probably be merged:
```suggestion
  Node* unique = pack->at(0)->in(j);
```